service/controller.go

```diff
@@ -287,6 +297,41 @@ func (s *service) CreateVolume(
 		// use the default if not set in the storage class
 		rootClientEnabled = RootClientEnabledParamDefault
 	}
+	//Setting Soft Limit
+	if _, ok := params[SoftLimitParam]; ok {
+		if params[SoftLimitParam] == "" {
+			softLimit = SoftLimitParamDefault
+		} else {
+			softLimit = params[SoftLimitParam]
+		}
+	} else {
+		// use the default if not set in the storage class
+		softLimit = SoftLimitParamDefault
+	}
+
+	// Setting Advisory Limit
+	if _, ok := params[AdvisoryLimitParam]; ok {
+		if params[AdvisoryLimitParam] == "" {
+			advisoryLimit = AdvisoryLimitParamDefault
+		} else {
+			advisoryLimit = params[AdvisoryLimitParam]
+		}
+	} else {
+		// use the default if not set in the storage class
+		advisoryLimit = AdvisoryLimitParamDefault
+	}
+
+	// Setting Soft Grace Period
+	if _, ok := params[SoftGracePrdParam]; ok {
```

Comment:
why this multiple if condition? can we refactor this?

Reply:
because all are independent parameters. So, checking if either of parameter is being passed read that value else set default value to that parameter.